lldb/include/lldb/Core/PluginManager.h

```diff
@@ -191,9 +191,7 @@ class PluginManager {
   GetObjectFileCreateMemoryCallbackForPluginName(llvm::StringRef name);
 
   static Status SaveCore(const lldb::ProcessSP &process_sp,
-                         const FileSpec &outfile,
-                         lldb::SaveCoreStyle &core_style,
-                         llvm::StringRef plugin_name);
+                         lldb_private::CoreDumpOptions &core_options);
```

Comment:
change to const: `const lldb_private::CoreDumpOptions &core_options`

Reply:
If we do this then we will need to remove all the current mutation of style in `MachO, ELF, Minidump`. (Or keep them as local variables, but in that case then it's confusing that a plugin could mutate your options and you woudln't know.

If we want to remove all the default settings per flavor in this patch let me know.